compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/RegExpParser.java

```diff
@@ -190,6 +190,13 @@ private STNode parseEscapeChar(STNode backSlash) {
             case PIPE_TOKEN:
                 return parseReQuoteEscape(backSlash);
             default:
+                if (nextToken.kind == SyntaxKind.RE_LITERAL_CHAR &&
```

Comment:
I think this case should be handled in `parseCharSetAtom()` method just before handling the `ReEscapes` in line no 431. If we handle this here it will affect the `ReEscapes`.
The following should result in a syntax error as `-` is not a `ReSyntaxChar`. but with this change it won't.
```ballerina
var t = re `\-`;
```